### PR TITLE
Set recaptcha params to Authentication context properties

### PR DESCRIPTION
**Pull request by shanggeeth**
## Purpose
Set recaptcha params in authentication context properties to be used to get redirect URI for Multi Option Page

## Related PRs
- https://github.com/wso2/carbon-identity-framework/pull/4301

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 1]
```diff
@@ -498,6 +498,9 @@ protected void processAuthenticationResponse(HttpServletRequest request,
                                                  HttpServletResponse response, AuthenticationContext context)
             throws AuthenticationFailedException {
 
+        context.setProperty("captchaParams",
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:501]: shall we make captchaParams a constant

Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:501]: Addressed in 737888e93ef15c373e5e0fe56126e3771e5be7da

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorConstants.java [hunk 2]
```diff
@@ -48,6 +48,7 @@ public abstract class BasicAuthenticatorConstants {
     public static final String RECAPTCHA_RESEND_CONFIRMATION_PARAM = "&reCaptchaResend=";
     public static final String RECAPTCHA_KEY_PARAM = "&reCaptchaKey=";
     public static final String RECAPTCHA_API_PARAM = "&reCaptchaAPI=";
+    public static final String CAPTCHA_PARAMS = "captchaParams";
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorConstants.java:51]: why can't we use the new constant that you have defined in [FrameworkConstants.java](https://github.com/wso2/carbon-identity-framework/pull/4301/files#diff-9461270f09d52c105a65affa1fdc9fbd740274afc25204ffd5ec2bf49e6975b9)?

**Review by sumedhe — DISMISSED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 3]
```diff
@@ -498,6 +498,9 @@ protected void processAuthenticationResponse(HttpServletRequest request,
                                                  HttpServletResponse response, AuthenticationContext context)
             throws AuthenticationFailedException {
 
+        context.setProperty(BasicAuthenticatorConstants.CAPTCHA_PARAMS,
+                getCaptchaParams(context.getLoginTenantDomain(), 0));
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:502]: better to set this to context only if `getCaptchaParams(context.getLoginTenantDomain(), 0)` not empty string as this is not needed to be set if recaptcha is disabled

Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:506]: Addressed in 0368728cbbd6a6d3edf8010df68c76ef4b9777ad

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorConstants.java (hunk 2 above)
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorConstants.java]: Addressed in 0368728cbbd6a6d3edf8010df68c76ef4b9777ad

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 4]
```diff
@@ -498,6 +498,13 @@ protected void processAuthenticationResponse(HttpServletRequest request,
                                                  HttpServletResponse response, AuthenticationContext context)
             throws AuthenticationFailedException {
 
+        String captchaParamString = getCaptchaParams(context.getLoginTenantDomain(), 0);
+        if (StringUtils.isNotBlank(captchaParamString)) {
+            context.setProperty(FrameworkConstants.CAPTCHA_PARAM_STRING, captchaParamString);
+        }
+        context.setProperty(FrameworkConstants.CAPTCHA_PARAM_STRING,
+                getCaptchaParams(context.getLoginTenantDomain(), 0));
```
[the remark this answers is not in the document]
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java]: Yes this is is unnecessary. Removed via 95f279af6588bdedd6c9cfc3eca8a2497efb2c06

**Review by dewniMW — APPROVED** [review]

**Pull request merged**